---
title: "tree_visualization"
output: html_document
---

#packages to load
```{r}
library(ggplot2)
library(treeio)
library(ggtree)
library(dplyr)
library(ape)
library(phytools)
library(readr)
library(seqinr)
library(xlsx)
library(readxl)
```

# dfs to load
```{r}
rooted_ref_tree_f = read.tree("/Users/rebeccachristensen/Desktop/Cremer_Lab_2022/main/workflow/out/raxmlOutputNovelFragInsert/RAxML_originalLabelledTree_noBootstrap_rooted_wKarthikSeqs.newick")

rooted_result_tree_f = read.tree("/Users/rebeccachristensen/Desktop/Cremer_Lab_2022/main/workflow/out/raxmlOutputNovelFragInsert/RAxML_labelledTree_noBootstrap_rooted_wKarthikSeqs.newick")

# closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef <- read_csv("/Users/rebeccachristensen/Desktop/Cremer_Lab_2022/main/workflow/out/treeInfo/closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef.csv")

closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef <- read_csv("closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef_wSingleClosestRef.csv")

otu_hits_table <- read_csv("otu_hits_table_wKarthikSeqs.csv", 
    col_types = cols(...1 = col_skip()))
```

# For hit sequences with multiple "closest ref sequences", select a single reference sequence as the representetive. Choose the first of Albert's sequences in the list. If none of Albert's sequences are in the closest ref list, choose the first of Karthik's sequences.
```{r}
get_single_ref <- function(ref_list) {
  ref_list <- gsub("\\]", "", ref_list)
  ref_list <- gsub("\\[", "", ref_list)
  ref_list <- gsub(" ", "", ref_list)
  ref_list <- gsub("\\'", "", ref_list)
  ref_list <- strsplit(ref_list, ",")[[1]]
  
  # check if all of the closest ref sequences are Karthik seqs
  if (length(grep("QUERY", ref_list)) == length(ref_list)) {
    # single ref is assigned as first seq in list
    return(ref_list[1])
  }
  
  # case if there is a non-Karthik seq in the list of closest_ref
  if (length(grep("QUERY", ref_list)) != length(ref_list)) {
    single_ref_index <- grep("QUERY", ref_list, invert = T)[[1]]
    return(ref_list[single_ref_index])
  }
}

closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef <- closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef %>%
  rowwise() %>%
  mutate(single_closest_ref = get_single_ref(closest_ref), na.rm = T)

# HA
# i am a god

write.csv(closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef, "closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef_wSingleClosestRef.csv")
```

# making a new otu df for each sequence in the ref tree, including Karthik seqs
```{r}
# read in the fasta file for ref seqs, complete with taxa descriptions from Albert's paper

# hack way for reading in a fasta file
ref_fa <- read_delim("muller_reference_seqs_with_taxa.fa.txt", delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

ref_fa <- na.omit(ref_fa) #lol it worked!
ref_fa <- rename(ref_fa, seq_abbreviation = X1,
                 seq_name = X2,
                 strain_name = X3,
                 environment = X4,
                 description = X5)

id <- as.character(ref_fa[1, "description"])
strsplit(id, ";")

get_taxa <- function(description, level) {
  taxa <- strsplit(as.character(description), ";")[[1]][level]
  return(taxa)
}

ref_fa <- ref_fa %>%
  rowwise() %>%
  mutate(tax_1 = get_taxa(description, 1),
         tax_2 = get_taxa(description, 2),
         tax_3 = get_taxa(description, 3),
         tax_4 = get_taxa(description, 4),
         tax_5 = get_taxa(description, 5))
# WEEEE

# ok i can bind the otu_hits_table to ref_fa using the seq_abbreviation column in ref_fa by creating a new column that matches this format in otu_hits_table

make_seq_abbreviation <- function(label) {
  label <- strsplit(as.character(label), "_")[[1]]
  abbreviation <- tail(label, n=1)
  abbreviation <- paste(">", abbreviation, sep="")
  
  return(abbreviation)
}

ref_seqs <- data_frame(tip_label = rooted_ref_tree_f$tip.label)
all_seqs <- data_frame(tip_label = rooted_result_tree_f$tip.label)

ref_seqs <- ref_seqs %>%
  rowwise() %>%
  mutate(seq_abbreviation = make_seq_abbreviation(tip_label))

# manually correct seq_abbreviation for Candidatus_Desulfovibrio_trichonymphae
ref_seqs$seq_abbreviation[ref_seqs$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$seq_abbreviation[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
ref_tax <- merge(ref_fa, ref_seqs, by = "seq_abbreviation")

karthik_seqs <- read_delim("/Users/rebeccachristensen/Desktop/Cremer_Lab_2022/main/config/Karthik_novel_seqs/Karthik2018_dsrAB_novelseqs.faa", delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
karthik_seqs <- subset(karthik_seqs, grepl(">", X1))
karthik_seqs <- rename(karthik_seqs, seq_abbreviation = X1)

make_tip_label <- function(seq_name){
  label <- gsub(">", "", as.character(seq_name))
  label <- paste("QUERY___", label, sep = "")
  return(label)
}

karthik_seqs <- karthik_seqs %>%
  rowwise() %>%
  mutate(tip_label = make_tip_label(seq_abbreviation))

# bind taxa data to df with tip labels
tax_df <- merge(karthik_seqs, ref_tax, by = c("tip_label", "seq_abbreviation"), all = T)

tax_df <- tax_df %>%
  rowwise() 
# %>%
#   mutate(tax_1 = ifelse(tip_label %in% karthik_seqs_in_tree$tip_label, "karthik", tax_1),
#          tax_2 = ifelse(tip_label %in% karthik_seqs_in_tree$tip_label, "karthik", tax_2),
#          tax_3 = ifelse(tip_label %in% karthik_seqs_in_tree$tip_label, "karthik", tax_3),
#          tax_4 = ifelse(tip_label %in% karthik_seqs_in_tree$tip_label, "karthik", tax_4),
#          tax_5 = ifelse(tip_label %in% karthik_seqs_in_tree$tip_label, "karthik", tax_5))

tax_df[is.na(tax_df)] <- " Unclassified"

karthik_seqs_in_tree <- subset(karthik_seqs, tip_label %in% all_seqs$tip_label)
write.csv

write.csv(tax_df, "tax_df, csv")
write.csv(ref_seqs, "ref_seqs.csv")
```

```{r}
# updated taxonomy info for named karthik seqs in tax_df
tax_df$tax_1[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrA"] <- tax_df$tax_1[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]
tax_df$tax_2[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrA"] <- tax_df$tax_2[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]
tax_df$tax_3[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrA"] <- tax_df$tax_3[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]
tax_df$tax_4[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrA"] <- tax_df$tax_4[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]
tax_df$tax_5[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrA"] <- tax_df$tax_5[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]

tax_df$tax_1[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrB"] <- tax_df$tax_1[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]
tax_df$tax_2[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrB"] <- tax_df$tax_2[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]
tax_df$tax_3[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrB"] <- tax_df$tax_3[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]
tax_df$tax_4[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrB"] <- tax_df$tax_4[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]
tax_df$tax_5[tax_df$tip_label == "QUERY___Bilophila_wadsworthia_dsrB"] <- tax_df$tax_5[tax_df$tip_label == "Bilophila_wadsworthia__BilWads6"]

tax_df$tax_1[tax_df$tip_label == "QUERY___Desulfosporosinus_orientis_dsrA"] <- tax_df$tax_1[tax_df$tip_label == "Desulfosporosinus_orientis__DesOrie4"]
tax_df$tax_2[tax_df$tip_label == "QUERY___Desulfosporosinus_orientis_dsrA"] <- tax_df$tax_2[tax_df$tip_label == "Desulfosporosinus_orientis__DesOrie4"]
tax_df$tax_3[tax_df$tip_label == "QUERY___Desulfosporosinus_orientis_dsrA"] <- tax_df$tax_3[tax_df$tip_label == "Desulfosporosinus_orientis__DesOrie4"]
tax_df$tax_4[tax_df$tip_label == "QUERY___Desulfosporosinus_orientis_dsrA"] <- tax_df$tax_4[tax_df$tip_label == "Desulfosporosinus_orientis__DesOrie4"]
tax_df$tax_5[tax_df$tip_label == "QUERY___Desulfosporosinus_orientis_dsrA"] <- tax_df$tax_5[tax_df$tip_label == "Desulfosporosinus_orientis__DesOrie4"]

tax_df$tax_1[tax_df$tip_label == "QUERY___Archaeoglobus_sulfaticallidus_dsrA"] <- tax_df$tax_1[tax_df$tip_label == "Archaeoglobus_sulfaticallidus__ArgSulfa"]
tax_df$tax_2[tax_df$tip_label == "QUERY___Archaeoglobus_sulfaticallidus_dsrA"] <- tax_df$tax_2[tax_df$tip_label == "Archaeoglobus_sulfaticallidus__ArgSulfa"]
tax_df$tax_3[tax_df$tip_label == "QUERY___Archaeoglobus_sulfaticallidus_dsrA"] <- tax_df$tax_3[tax_df$tip_label == "Archaeoglobus_sulfaticallidus__ArgSulfa"]
tax_df$tax_4[tax_df$tip_label == "QUERY___Archaeoglobus_sulfaticallidus_dsrA"] <- tax_df$tax_4[tax_df$tip_label == "Archaeoglobus_sulfaticallidus__ArgSulfa"]
tax_df$tax_5[tax_df$tip_label == "QUERY___Archaeoglobus_sulfaticallidus_dsrA"] <- tax_df$tax_5[tax_df$tip_label == "Archaeoglobus_sulfaticallidus__ArgSulfa"]

tax_df$tax_1[tax_df$tip_label == "QUERY___Thermodesulfobium_narugense_dsrA"] <- tax_df$tax_1[tax_df$tip_label == "Thermodesulfobium_narugense__TrfNaru2"]
tax_df$tax_2[tax_df$tip_label == "QUERY___Thermodesulfobium_narugense_dsrA"] <- tax_df$tax_2[tax_df$tip_label == "Thermodesulfobium_narugense__TrfNaru2"]
tax_df$tax_3[tax_df$tip_label == "QUERY___Thermodesulfobium_narugense_dsrA"] <- tax_df$tax_3[tax_df$tip_label == "Thermodesulfobium_narugense__TrfNaru2"]
tax_df$tax_4[tax_df$tip_label == "QUERY___Thermodesulfobium_narugense_dsrA"] <- tax_df$tax_4[tax_df$tip_label == "Thermodesulfobium_narugense__TrfNaru2"]
tax_df$tax_5[tax_df$tip_label == "QUERY___Thermodesulfobium_narugense_dsrA"] <- tax_df$tax_5[tax_df$tip_label == "Thermodesulfobium_narugense__TrfNaru2"]

tax_df$seq_abbreviation[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$seq_abbreviation[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$seq_name[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$seq_name[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$strain_name[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$strain_name[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$environment[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$environment[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$description[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$description[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$tax_1[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$tax_1[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$tax_2[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$tax_2[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$tax_3[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$tax_3[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$tax_4[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$tax_4[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]
tax_df$tax_5[tax_df$tip_label == "Candidatus_Desulfovibrio_trichonymphae"] <- ref_fa$tax_5[ref_fa$seq_name == "Candidatus Desulfovibrio trichonymphae"]

```

# otu_hits_table: number of hit sequences (including duplicates) associated with each reference sequence (by shortest branch distance) for all samples
# sample_hits_table: number of samples ("Biosample" as defined by NCBI) containing each hit sequence (rounded to shortest reference sequence)
```{r}
otu_hits_table <- as.data.frame(table(subset(closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef$single_closest_ref, closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef$distance < 2.5 | is.na(closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef$distance))))
otu_hits_table <- rename(otu_hits_table, tip_label = Var1)

write.csv(otu_hits_table, "otu_hits_table_wKarthikSeqs.csv")

sample_hits_table <- closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef %>%
  filter(distance < 2.5 | is.na(distance)) %>%
  group_by(single_closest_ref) %>%
  summarize(n_samples = n_distinct(biosample_id))

sample_hits_table <- rename(sample_hits_table, tip_label = single_closest_ref)

write.csv(sample_hits_table, "sample_hits_table_wKarthikSeqs.csv")



sample_hits_long <- closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef %>%
  filter(distance < 2.5 | is.na(distance)) %>%
  left_join(., phylogeny_df, join_by(single_closest_ref == tip_label)) %>%
  select(biosample_id, seq_name, tax_1:tax_5, phylum:family) %>%
  pivot_longer(c(tax_1:tax_5, phylum:family), names_to = "rank", values_to = "taxon") %>%
  group_by(taxon, rank, biosample_id) %>%
  summarize(n_hits = n())
  
write.csv(sample_hits_long, "hits_per_sample_by_taxa_long.csv")

```

# trim results tree to only include ref seqs and novel karthik seqs
```{r}
ref_tips <- append(ref_seqs$tip_label, karthik_seqs_in_tree$tip_label)
query_tips <- setdiff(all_seqs$tip_label, ref_tips)

pruned <- drop.tip(rooted_result_tree_f, query_tips)

# pruned_plot <- 
  ggtree(pruned, layout = "circular")

# function to auto generate list of groups for different taxon levels courtsey of the internet woohoo
df <- tax_df
df <- subset(df, tax_5 != "" & !is.na(tax_5) & !is.na(tip_label))

groupInfo <- split(df$tip_label, df$tax_5)

pruned_groups <- groupOTU(pruned, groupInfo)

pruned_plot <-
  ggtree(pruned_groups, aes(color = group))

pruned_plot %<+%
  otu_hits_table[,c("tip_label", "Freq")] +
  aes(color = Freq) +
  scale_color_gradient(low = "blue", high = "orange") +
  theme(legend.position = "")

ggplot(otu_hits_table, aes(x = Freq, y = tip_label)) + 
  geom_bar(stat = "identity")

# closest hits only tree
not_closest_hits <- setdiff(rooted_result_tree_f$tip.label, otu_hits_table$tip_label)
closest_hits_tree <- drop.tip(rooted_result_tree_f, not_closest_hits)

tax_df_closest_hits <- subset(tax_df, tip_label %in% closest_hits_tree$tip.label)

# manually checking phylogeny for sequences in closest_hits_tree; NCBI has updates taxa info by the seq names!

write.xlsx(tax_df_closest_hits, "tax_df_closest_hits_original.xlsx")
tax_df_closest_hits <- read_excel("tax_df_closest_hits.xlsx", 
                                                         sheet = "Sheet2", col_types = c("skip", 
                                                                                         "text", "text", "text", "text", "text", 
                                                                                         "text", "text", "text", "text", "text", 
                                                                                         "text", "text"))

tax_df_closest_hits <- tax_df_closest_hits %>%
  mutate(source = ifelse(grepl("QUERY", tip_label), 
                         "Anantharaman 2018", 
                         "Müller 2015"))
  
dsrAB_tree_with_bar <- ggtree(groupOTU(closest_hits_tree, groupInfo), aes(fill = group, color = group)) %<+%
  tax_df_closest_hits +
  geom_tiplab(size = 2.5, aes(label = names_for_tree), offset = 0.1) +
  geom_tippoint(aes(shape = source), size = 2) +
  xlim_tree(5) +
  geom_facet(data = data.frame(otu_hits_table), panel = "N Hits", geom = geom_bar, mapping = (aes(x = Freq)), orientation = "y", stat = "identity") +
  geom_facet(data = subset(data.frame(sample_hits_table), !is.na(single_closest_ref)), panel = "N Samples", geom = geom_bar, mapping = (aes(x = n_samples)), orientation = "y", stat = "identity") +
guides(fill = guide_legend(title = "family-level dsrAB taxon", nrow = 3, title.position = "top"), color = "none", shape = guide_legend(title = "sequence source", nrow = 2, title.position = "top")) +
  theme_classic() +
  theme(legend.position = "bottom", 
        legend.text = element_text(size = 8),
        legend.title = element_text(size = 9),
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank())
  
# "N Hits" column units: number of HMMER hits including duplicates that pass score threshold associated with each reference sequence (shortest branch distance) 
# "N sample" units: number of biosamples that contained each reference seq (and all dups etc etc)
# Tree branch length units: mean number of substitutions / site
# (high bc of high variance between tips and root for this tree... very diverse indeed)
pdf("dsrAB_hits_tree_with_barchart.pdf", width = 9)

facet_widths(dsrAB_tree_with_bar, widths = c(3, 1, 1))

```

# retrieve data on 16S rRNA phylogeny for closest ref sequences
```{r}
# code from get_taxonomy.Rmd
library(taxize)
library(myTAI)

müller_exception_strain_names <- c("uncultured bacterium", "Bioreactor clone", "Polluted aquifer clone", "Termite protist endosymbiont community metagenome")

müller_strains <- unique(subset(tax_df_closest_hits, source == "Müller 2015" & !(seq_name %in% müller_exception_strain_names))$seq_name)
müller_phylogeny <- data.frame(seq_name = müller_strains)

for (seq_name in müller_strains) {
  lineage <- taxonomy(organism = seq_name, db = "ncbi", output = "classification")
  if ("phylum" %in% lineage$rank) {
    phylum <- lineage$name[lineage$rank == "phylum"]
    müller_phylogeny$phylum[müller_phylogeny$seq_name == seq_name] <- phylum
  }

  if ("class" %in% lineage$rank) {
    class <- lineage$name[lineage$rank == "class"]
    müller_phylogeny$class[müller_phylogeny$seq_name == seq_name] <- class
  }

  if ("order" %in% lineage$rank) {
    order <- lineage$name[lineage$rank == "order"]
    müller_phylogeny$order[müller_phylogeny$seq_name == seq_name] <- order
  }

  if ("family" %in% lineage$rank) {
    family <- lineage$name[lineage$rank == "family"]
    müller_phylogeny$family[müller_phylogeny$seq_name == seq_name] <- family
  }
}

müller_phylogeny_merged <- merge(subset(tax_df_closest_hits, source == "Müller 2015" & !(seq_name %in% müller_exception_strain_names)), müller_phylogeny, by = "seq_name")

anantharam_strains <- unique(subset(tax_df_closest_hits, source == "Anantharaman 2018" | seq_name %in% müller_exception_strain_names)$names_for_tree)
anantharam_phylogeny <- data.frame(names_for_tree = anantharam_strains)

for (names_for_tree in anantharam_strains) {
  lineage <- taxonomy(organism = names_for_tree, db = "ncbi", output = "classification")
  if ("phylum" %in% lineage$rank) {
    phylum <- lineage$name[lineage$rank == "phylum"]
    anantharam_phylogeny$phylum[anantharam_phylogeny$names_for_tree == names_for_tree] <- phylum
  }
  
  if ("class" %in% lineage$rank) {
    class <- lineage$name[lineage$rank == "class"]
    anantharam_phylogeny$class[anantharam_phylogeny$names_for_tree == names_for_tree] <- class
  }

  if ("order" %in% lineage$rank) {
    order <- lineage$name[lineage$rank == "order"]
    anantharam_phylogeny$order[anantharam_phylogeny$names_for_tree == names_for_tree] <- order
  }

  if ("family" %in% lineage$rank) {
    family <- lineage$name[lineage$rank == "family"]
    anantharam_phylogeny$family[anantharam_phylogeny$names_for_tree == names_for_tree] <- family
  }
}

anantharam_phylogeny_merged <- merge(subset(tax_df_closest_hits, source == "Anantharaman 2018" | seq_name %in% müller_exception_strain_names), anantharam_phylogeny, by = "names_for_tree")

phylogeny_df <- rbind(anantharam_phylogeny_merged, müller_phylogeny_merged)

phylogeny_df[is.na(phylogeny_df)] <- "Unclassified"

phylogeny_df <- left_join(phylogeny_df, otu_hits_table, by = "tip_label")

write.csv(phylogeny_df, "16sTaxonomy_closestRef.csv")

```

# 16S rRNA phylogeny visualization -- bar charts for frequency of different taxa
```{r}
phylogeny_df_long <- phylogeny_df %>%
  select(phylum:Freq) %>%
  pivot_longer(phylum:family, names_to = "rank", values_to = "taxon") %>%
  group_by(taxon, rank) %>%
  summarize(freq = sum(Freq))

phylogeny_df_long$rank <- factor(phylogeny_df_long$rank, levels = c("family", "order", "class", "phylum"))

ggplot(phylogeny_df, aes(x =  Freq, y = family)) +
  geom_bar(stat = "identity")

phylogeny_barplot <- ggplot(phylogeny_df_long, aes(x = freq, y = reorder(taxon, freq))) +
  geom_bar(stat = "identity") +
  facet_wrap(~rank, scales = "free_y", ncol = 1) +
  ylab("Taxon") +
  xlab("N Hit Sequences") +
  ggtitle("N Hit Sequences by 16S rRNA Phylogeny") +
  scale_x_continuous(expand = c(0, 0)) +
  theme_bw()
  
pdf("phylogeny_barchart.pdf", height = 7, width = 6)
phylogeny_barplot

```

# by-sample phylogeny
```{r}
sample_hits_long <- read_csv("hits_per_sample_by_taxa_long.csv")

nhits_histogram <- ggplot(subset(sample_hits_long, rank == "tax_1"), aes(x = n_hits)) +
  geom_histogram(binwidth = 1, fill = "grey", color = "black") +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_continuous(expand = c(0, 0)) +
  theme_bw() +
  xlab("N Hits") +
  ylab("N Samples") +
  ggtitle("Distribution of N Hits Across All Samples")

samples <- unique(subset(sample_hits_long, rank == "tax_1" & n_hits > 5)$biosample_id)
palette <- c("#3d3a39", "#ee88af", "#81c1cb", "#f8f3ed", "#f3de09")

phylum_prop_plot <- ggplot(subset(sample_hits_long, rank == "phylum" & biosample_id %in% samples & !is.na(taxon)), aes(x = biosample_id, y = n_hits, fill = taxon)) + 
  geom_bar(position = "fill", stat = "identity") +
  scale_fill_manual(values = palette) +
  theme_classic() +
  guides(fill = guide_legend(title.position = "top", title = "Phylum")) +
  ylab("Proportion of Hits") +
  xlab("Sample (subset for > 5 total hits)") +
  ggtitle("Proportion of Phyla per Sample") +
  scale_y_continuous(expand = c(0, 0)) +
  theme(legend.position = "bottom", 
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank())

family_prop_plot <- ggplot(subset(sample_hits_long, rank == "family" & biosample_id %in% samples & !is.na(taxon)), aes(x = biosample_id, y = n_hits, fill = taxon)) + 
  geom_bar(position = "fill", stat = "identity") +
  scale_fill_manual(values = brewer.pal(7, "Set2")) +
  theme_classic() +
  guides(fill = guide_legend(title.position = "top", title = "Family")) +
  ylab("Proportion of Hits") +
  xlab("Sample (subset for > 5 total hits)") +
  ggtitle("Proportion of Families per Sample") +
  scale_y_continuous(expand = c(0, 0)) +
  theme(legend.position = "bottom", 
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank())

pdf("n_hits_histogram.pdf", height = 5)
nhits_histogram
dev.off()

pdf("phylum_proportions_by_sample.pdf", height = 5)
phylum_prop_plot
dev.off()

pdf("family_proportions_by_sample.pdf", height = 5)
family_prop_plot
```



# I need to check the math on this... what does log-transforming change in the representation and interpretation of the data? 
```{r}

otu_hits_table <- as.data.frame(table(subset(closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef$single_closest_ref, closestRefInfo_allScoreThresholdHits_wKarthikSeqsAsRef$distance < 2.5)))

otu_hits_table$logFreq <- log2(otu_hits_table$Freq + 1)

otu_hits_table <- rename(otu_hits_table, single_closest_ref = Var1)

hit_otus <- otu_hits_table$single_closest_ref
nonhit_otus <- setdiff(rooted_ref_tree$tip.label, hit_otus)

tobind_df <- data.frame(single_closest_ref = nonhit_otus, Freq = 0, logFreq = 0)

otu_all_table <- rbind(otu_hits_table, tobind_df)

otu_all_table$logFreq <- log2(otu_all_table$Freq + 1)

rooted_ref_tree_plot <- ggtree(rooted_ref_tree_f, layout = "circular")

rooted_ref_tree_plot %<+%
  otu_hits_table +
  aes(color = Freq) +
  scale_color_gradient(low = "blue", high = "orange") +
  theme(legend.position = "")

rooted_ref_tree_plot <- ggtree(rooted_ref_tree_f)

rooted_ref_tree_plot %<+%
  otu_hits_table +
  aes(color = Freq) +
  scale_color_gradient(low = "blue", high = "orange") +
  geom_bar(aes(x = Freq), stat = "identity")

p + geom_facet(data = otu_hits_table, geom = geom_col, 
                aes(x = Freq), orientation = 'y', width = .6, panel = "wee")

ggplot(otu_hits_table, aes(x = Freq, y = Var1)) + 
  geom_bar(stat = "identity")

data(iris)
rn <- paste0(iris[,5], "_", 1:150)
rownames(iris) <- rn
d_iris <- dist(iris[,-5], method="man")

tree_iris <- ape::bionj(d_iris)
grp <- list(setosa     = rn[1:50],
            versicolor = rn[51:100],
            virginica  = rn[101:150])

```

# heatmap of entire tree (circular)
```{r}

rooted_ref_tree_plot <- ggtree(rooted_ref_tree, layout = "circular") + geom_tiplab(size = 0)

rooted_ref_tree_plot %<+%
  otu_hits_table +
  aes(color = Freq) +
  scale_color_gradient(low = "blue", high = "orange")

hits_only_tree <- keep.tip(rooted_ref_tree_plot, c(hits_list, ox_bact_node, nitrospirae, red_arch_node))
hits_only_tree <- ggtree(hits_only_tree) + xlim_tree(15)

```

# create heatmap by clade
```{r}

# gheatmap(rtree, data= otu_subj_mat, offset = -0.35, width = 0.5, legend_title = "Number of Hits \n (log plus one)", colnames = FALSE, low = "blue", high = "orange", color = "transparent") +
#   labs(fontsize = 4) +
#   theme(legend.position = c(1, 0.25), legend.key.size = unit(0.45, "cm"))

desulfovibrionaceae_node = MRCA(rooted_ref_tree, c('Mariager_Fjord_clone_MAF65G__Af2Fjo18', "Bioreactor_clone__UncS2021"))
viewClade(rooted_ref_tree, desulfovibrionaceae_node)

firmicutes_node = MRCA(rooted_ref_tree, 'Bioreactor_clone__UncS2026', "Black_Sea_sediment_clone_BSI-11__BlackS15")
viewClade(rooted_ref_tree, firmicutes_node)
```

# nodes
```{r}
hits_only_tree <- keep.tip(rooted_ref_tree, c(hits_list, ox_bact_node, nitrospirae, red_arch_node))

ox_bact_node <- intersect(as.vector(subset(taxa_table$OTU, taxa_table$dsrab_type == "Oxidative bacterial type DsrAB")), rooted_ref_tree$tip.label)

nitrospirae <- intersect(as.vector(subset(taxa_table$OTU, taxa_table$supercluster == "Nitrospirae supercluster")), rooted_ref_tree$tip.label)

red_arch_node <- intersect(as.vector(subset(taxa_table$OTU, taxa_table$dsrab_type == "Reductive archaeal type DsrAB")), rooted_ref_tree$tip.label)

bilophila_node <- as.vector(subset(taxa_table$OTU, grepl("Bilophila", taxa_table$OTU) == TRUE))

dpiger_node <- as.vector(subset(taxa_table$OTU, grepl("Desulfovibrio_piger", taxa_table$OTU) == TRUE))

dalkalitolerans_node <- as.vector(subset(taxa_table$OTU, grepl("Desulfitibacter_alkalitolerans", taxa_table$OTU) == TRUE))

firmicutes_node <- intersect(as.vector(subset(taxa_table$OTU, taxa_table$supercluster == "FACA group sensu lato")), hits_only_tree$tip.label)

desulfovibrionaceae_node <- intersect(as.vector(subset(taxa_table$OTU, taxa_table$level3_taxon == "Desulfovibrionaceae")), hits_only_tree$tip.label)

gpam_node <- intersect(as.vector(subset(taxa_table$OTU, taxa_table$level3_taxon == "Gordonibacter")), hits_only_tree$tip.label)

dvulgaris_node <- intersect(as.vector(subset(taxa_table$OTU, grepl("Desulfovibrio_vulgaris", taxa_table$OTU) == TRUE)), hits_only_tree$tip.label)

dcarboniphilus_node <- intersect(as.vector(subset(taxa_table$OTU, grepl("Desulfovibrio_carbinoliphilus", taxa_table$OTU) == TRUE)), hits_only_tree$tip.label)

tnarugense_node <- intersect(as.vector(subset(taxa_table$OTU, grepl("Thermodesulfobium", taxa_table$OTU) == TRUE)), hits_only_tree$tip.label)

archaeoglobus_node <- intersect(as.vector(subset(taxa_table$OTU, grepl("Archaeoglobus", taxa_table$OTU) == TRUE)), hits_only_tree$tip.label)

desulfitobacterium_node <- intersect(as.vector(subset(taxa_table$OTU, grepl("Desulfitobacterium", taxa_table$OTU) == TRUE)), hits_only_tree$tip.label)

```

# tree with labels for all hit ref leaves (for reference)
```{r}
hits_only_tree <- keep.tip(rooted_ref_tree, c(hits_list, ox_bact_node, nitrospirae, red_arch_node))
hits_only_tree <- ggtree(hits_only_tree) + xlim_tree(15) + geom_tiplab(size = 2, offset = 0.3)

tree <- hits_only_tree %>%
  # collapse and label oxidative bacterial node
  collapse(node = MRCA(hits_only_tree, ox_bact_node)) %>%
  collapse(node = MRCA(hits_only_tree, red_arch_node)) %>%
  collapse(node = MRCA(hits_only_tree, nitrospirae)) +
  
  # outgroup labels
  
  geom_point2(aes(subset=(node == MRCA(hits_only_tree, ox_bact_node))), shape = 21, fill = "gray", color = "gray") +
  geom_cladelabel(node = MRCA(hits_only_tree, ox_bact_node), label = "oxidative bacterial type dsrAB", fontsize = 2.5, color = "gray", offset = 0.3) +
  
  geom_point2(aes(subset=(node == MRCA(hits_only_tree, nitrospirae))), shape = 21, fill = "gray", color = "gray") +
  geom_cladelabel(node = MRCA(hits_only_tree, nitrospirae), label = "nitrospirae supercluster", fontsize = 2.5, color = "gray", offset = 0.3) +
  
  geom_point2(aes(subset=(node == MRCA(hits_only_tree, red_arch_node))), shape = 21, fill = "gray", color = "gray") +
  geom_cladelabel(node = MRCA(hits_only_tree, red_arch_node), label = "reductive archaeal type dsrAB", fontsize = 2.5, color = "gray", offset = 0.3) +
  
  # group labels
  
  geom_cladelabel(node = MRCA(hits_only_tree, firmicutes_node), label = "Firmicutes group", fontsize = 4, offset = 5.5, alpha = 0.5) +
  
  geom_cladelabel(node = MRCA(hits_only_tree, desulfovibrionaceae_node), label = "Desulfovibrionaceae", fontsize = 4, offset = 4.5, alpha = 0.5)

tree %<+%
  otu_all_table +
  geom_point2(aes(color = logFreq, subset = updated_freq > 0)) +
  scale_color_gradient(low = "blue", high = "orange") +
  # guides(color = "none", size = "none") +
  theme(legend.position = "left")

```

# make it pretty
```{r}

# rooted_ref_tree = ape::read.tree("/Users/rebeccachristensen/Desktop/Cremer_Lab_2022/main/workflow/out/raxmlOutputFinal/RAxML_originalLabelledTree_noBootstrap.newick")
# 
# # # only keep tips that have hits
hits_list <- as.vector(otu_hits_table$Var1)

hits_only_tree <- keep.tip(rooted_ref_tree, c(hits_list, ox_bact_node, nitrospirae, red_arch_node))

# hits_only_tree <- ggtree(hits_only_tree) + geom_tiplab(aes(subset=(label %in% otus_to_label)), size = 2) + xlim_tree(15)
# hits_only_tree <- ggtree(hits_only_tree) + geom_tiplab(aes(subset=(grepl("Desulfovibrio", label)) == TRUE), size = 2) + xlim_tree(15)

hits_only_tree <- 
  ggtree(hits_only_tree, color = "dark gray") + xlim_tree(10)


tree <- hits_only_tree %>%
  # collapse and label oxidative bacterial node
  collapse(node = MRCA(hits_only_tree, ox_bact_node)) %>%
  collapse(node = MRCA(hits_only_tree, red_arch_node)) %>%
  collapse(node = MRCA(hits_only_tree, nitrospirae)) +

  # outgroup labels

  geom_point2(aes(subset=(node == MRCA(hits_only_tree, ox_bact_node))), shape = 21, fill = "dark gray", color = "dark gray") +
  geom_cladelabel(node = MRCA(hits_only_tree, ox_bact_node), label = "oxidative bacterial type dsrAB", fontsize = 2.5, color = "dark gray") +

  geom_point2(aes(subset=(node == MRCA(hits_only_tree, nitrospirae))), shape = 21, fill = "dark gray", color = "dark gray") +
  geom_cladelabel(node = MRCA(hits_only_tree, nitrospirae), label = "nitrospirae supercluster", fontsize = 2.5, color = "dark gray") +

  geom_point2(aes(subset=(node == MRCA(hits_only_tree, red_arch_node))), shape = 21, fill = "dark gray", color = "dark gray") +
  geom_cladelabel(node = MRCA(hits_only_tree, red_arch_node), label = "reductive archaeal type dsrAB", fontsize = 2.5, color = "dark gray") +

  # geom_point2(aes(subset=(node == MRCA(hits_only_tree, "Moorella_thermoacetica_Y72_copy_2__MooTher5"))), shape = 21, fill = "dark gray", color = "dark gray") +
  # geom_cladelabel(node = MRCA(hits_only_tree, "Moorella_thermoacetica_Y72_copy_2__MooTher5"), label = "Moorella thermoacetica, dsrAB copy 2", fontsize = 2.5, offset.text = 0.3, color = "dark gray") +

  # species labels

  geom_cladelabel(node = MRCA(hits_only_tree, bilophila_node), label = 'paste(italic("Bilophila"), " spp.* 59.7%")', parse = T, fontsize = 2.5, color = "black") +

  geom_cladelabel(node = MRCA(hits_only_tree, dpiger_node), label = 'paste(italic("Desulfovibrio piger"), "* 16.9%")', parse = T, fontsize = 2.5, color = "black") +

  geom_cladelabel(node = MRCA(hits_only_tree, dalkalitolerans_node), label = 'paste(italic("Desulfitibacter alkalitolerans"), " 11.8%")', parse = T, fontsize = 2.5) +

  geom_cladelabel(node = MRCA(hits_only_tree, gpam_node), label = 'paste(italic("Gordonibacter pamelaeae"), "* 6.5%")', parse = T, fontsize = 2.5, color = "black") +
  geom_cladelabel(node = MRCA(hits_only_tree, dvulgaris_node), label = 'paste(italic("Desulfovibrio vulgaris"), "* 4.0%")', parse = T, fontsize = 2.5, color = "black") +

  geom_cladelabel(node = MRCA(hits_only_tree, dcarboniphilus_node), label = 'paste(italic("Desulfovibrio carboniphilus"), " <1%")', parse = T, fontsize = 2.5) +

  geom_cladelabel(node = MRCA(hits_only_tree, tnarugense_node), label = 'paste(italic("Thermodesulfobium narugense"), " 1.2%")', parse = T, fontsize = 2.5) +

  geom_cladelabel(node = MRCA(hits_only_tree, archaeoglobus_node), label = 'paste(italic("Archaeoglobus sulfaticallidus"), " 1.3%")', parse = T, fontsize = 2.5) +

  geom_cladelabel(node = MRCA(hits_only_tree, desulfitobacterium_node), label = 'paste(italic("Desulfitobacterium hafniense"), " <1%")', parse = T, fontsize = 2.5) +

  geom_cladelabel(node = MRCA(hits_only_tree, "Desulfovibrio_simplex__DslSimpl"), label = 'paste(italic("Desulfovibrio simplex"), " <1%")', parse = T, fontsize = 2.5) +

  geom_cladelabel(node = MRCA(hits_only_tree, "Candidatus_Desulfovibrio_trichonymphae"), label = 'paste(italic("Candidatus"), " Desulfovibrio trichonymphae <1%")', parse = T, fontsize = 2.5) +

  geom_cladelabel(node = MRCA(hits_only_tree, "Desulfosporosinus_youngiae__DfpYoun5"), label = 'paste(italic("Desulfosporosinus youngiae"), " <1%")', parse = T, fontsize = 2.5) +

  geom_cladelabel(node = MRCA(hits_only_tree, c("Desulfovibrio_sp._6_1_46AFAA__DsvSp231", "Desulfovibrio_sp._3_1_syn3__DsvSp230", "Desulfovibrio_sp._Dsv1__DsvSp233")), label = 'paste(italic("Desulfovibrio"), " spp. 3.8%")', parse = T, fontsize = 2.5) +

  # Desulfovibrio desulfuricans labels
  geom_cladelabel(node = MRCA(hits_only_tree, "Desulfovibrio_desulfuricans__DQ092635"), label = 'paste(italic("Desulfovibrio desulfuricans*"), " <1%")', parse = T, fontsize = 2.5, color = "black") +
  geom_cladelabel(node = MRCA(hits_only_tree, c("Desulfovibrio_desulfuricans__DslDes17", "Desulfovibrio_desulfuricans_subsp._desulfuricans_str._ATCC_27774__CP001358")), label = 'paste(italic("Desulfovibrio desulfuricans*"), " <1%")', parse = T, fontsize = 2.5, color = "black") +
  geom_cladelabel(node = MRCA(hits_only_tree, "Desulfovibrio_desulfuricans__DslDes16_DsvAlas8"), label = 'paste(italic("Desulfovibrio desulfuricans*"), " <1%")', parse = T, fontsize = 2.5, color = "black") +
  geom_cladelabel(node = MRCA(hits_only_tree, "Desulfovibrio_desulfuricans__DslDes24"), label = 'paste(italic("Desulfovibrio desulfuricans*"), " <1%")', parse = T, fontsize = 2.5, color = "black") +

  # group labels

  # geom_cladelabel(node = MRCA(hits_only_tree, firmicutes_node), label = "Firmicutes group s.l. ", fontsize = 2.5, offset = 4, horizontal = FALSE) +
  #
  # geom_cladelabel(node = MRCA(hits_only_tree, desulfovibrionaceae_node), label = "Desulfovibrionaceae", fontsize = 2.5, offset = 3, horizontal = FALSE)

    geom_cladelabel(node = MRCA(hits_only_tree, firmicutes_node), label = "", fontsize = 2.5, offset = 4, horizontal = FALSE) +

  geom_cladelabel(node = MRCA(hits_only_tree, desulfovibrionaceae_node), label = "", fontsize = 2.5, offset = 3, horizontal = FALSE)


# tree %<+%
#   otu_all_table +
#   geom_point2(aes(color = logFreq, subset = Freq > 0)) +
#   scale_color_gradient(low = "orange", high = "blue") +
#   # guides(color = "none") +
#   theme(legend.position = "left")


```

# aesthetic mappings for the tree
```{r}
tree <-
  tree %<+%
  otu_all_table +
  geom_point2(aes(color = logFreq, subset = updated_freq > 0)) +
  scale_color_gradient(low = "orange", high = "blue", name = "") +
  # guides(color = "none") +
  theme(legend.position = c(0.09, 0.85), legend.key.size = unit(0.5, "cm"),legend.title = element_text(size=6)) +
  geom_treescale(x = 0.09, y = 30, width = 0.5, fontsize = 3, color = "darkgray")
  

```

# adding facet panel for frequency
```{r}
# facet_plot(tree, panel="N hits", data=subset(otu_all_table, otu_all_table$Var1 %in% otu_hits_table$Var1), geom_col, mapping = aes(x=updated_freq))

# subset(otu_all_table, Var1 %in% otu_hits_table$Var1)
frequency_barplot <-
ggplot(data = subset(otu_all_table, Var1 %in% otu_hits_table$Var1)) +
  geom_col(mapping = aes(x=updated_freq, y = Var1))+
  scale_x_reverse() +
  # theme(axis.text.y=element_blank(),
  #       axis.ticks.y=element_blank(),
  #       axis.title.y=element_blank()) +
  xlab("n Hits")
#   
# 
# tree_and_bar <- 
tree + frequency_barplot
# facet_widths(tree_and_bar, widths = c(1, 2))

```

# label leaves w/ unclassified bacteria on tree
```{r}
# unclassified_nodes <- c("Rice_field_soil_clone_Bu-23__JN615148", "uncultured_bacterium__EF645666", "uncultured_bacterium__EF645667", "uncultured_bacterium__EF645664", "uncultured_bacterium__EF645665", "uncultured_bacterium__EF645666", "Bioreactor_clone__UncS2028", "Termite_protist_endosymbiont_community_metagenome__entry118", "Polluted_aquifer_clone__UncS1623", "MSW_digester_clone__Unc01cxc", "Anaerobic_bacterium_sk.prop8__Bv2Prop2", "Anaerobic_bacterium_EtOH8__Bv2EtOH2", "Deep_sea_sediment_clone__JN798931", "Bioreactor_clone__UncS2029", "Arctic_sediment_clone_AH5_5__AtcSed67", "Pearl_river_estuary_clone_TopDSR14__UncS2036", "Polluted_aquifer_clone_LGWK16__UncS1633")

for (node in unclassified_nodes) {
  tree <- tree +  geom_cladelabel(node = MRCA(hits_only_tree, node), label = "unclassified bacteria", fontsize = 2, color = "dark grey")
}

tree
```

# tree w/ all hits
```{r}
# you got this!
# "you go girl" -Zack

# first, let's visualize the tree with all closest reference leaves and hit sequences

# load result tree in and root

# result_tree = read.tree("/Users/rebeccachristensen/Desktop/Cremer_Lab_2022/main/workflow/out/raxmlOutputFinal/RAxML_labelledTree_noBootstrap_rooted.newick")
# rooted_result_tree <-root(result_tree, c("Moorella_thermoacetica_Y72_copy_2__MooTher5"), r = TRUE)
# # rooted_result_tree <- ggtree(rooted_result_tree, layout = "circular")
# 
# 
# # only keep leaves that are in the closest ref leaves list OR have "query" in the name
# closest_ref_leaves <- as.vector(otu_hits_table$Var1)  # closest ref leaves from otu_hits_table (made from the ref tree)
# query_leaves <- as.vector(subset(rooted_result_tree$tip.label, grepl("QUERY", rooted_result_tree$tip.label) == TRUE))
# trimmed_result_tree <- keep.tip(rooted_result_tree, c(closest_ref_leaves, query_leaves))
# trimmed_result_tree <- drop.tip(trimmed_result_tree, c(moorella_queries, "Moorella_thermoacetica_Y72_copy_2__MooTher5"))

# collapse Bilophila queries


tree <- tree_subset(trimmed_result_tree, "Bilophila_wadsworthia__BilWads6", levels_back = 4)

tree <- ggtree(tree, layout = "circular") + 
  geom_tiplab2(aes(subset= label %in% closest_ref_leaves), size = 2)
  # geom_tiplab()

tree +
  # collapse(node = MRCA(tree, bilophila_node)) %>%
  # collapse(node = MRCA(tree, dpiger_node)) +
  # collapse(node = MRCA(tree, c("QUERY___S1768", "QUERY___S2"))) +
  # collapse(node = MRCA(tree, c("QUERY___S9", "QUERY___S1286"))) +
  # collapse(node = MRCA(tree, c("QUERY___S9", "QUERY___S1286"))) +

  geom_hilight(node = MRCA(tree, bilophila_node)) +
  geom_cladelabel(node = MRCA(tree, bilophila_node), label = "Bilophila", size = 2) +
  geom_cladelabel(node = MRCA(tree, dpiger_node), label = "D. piger", size = 2)

# tree %<+%
#   otu_all_table +
#   geom_point2(aes(color = updated_freq, subset = updated_freq > 0)) +
#   scale_color_gradient(low = "blue", high = "orange") +
#   # guides(color = "none") +
#   theme(legend.position = c(0.1, 0.85))

```

# scoring system for hits with multiple closest reference sequences
```{r}
otu_hits_table <- as.data.frame(table(subset(closestRefInfo_allScoreThresholdHits_singleRefOnly$closest_ref, closestRefInfo_allScoreThresholdHits_singleRefOnly$distance < 2.5)))

# update frequencies for multiple hits
for (otu_list in otu_hits_table$Var1) {
  list_freq <- otu_hits_table$Freq[otu_hits_table$Var1 == otu_list]
  otu_list <- gsub("\\[|\\]|\\'|\\ ", "", otu_list)
  otu_list <- strsplit(otu_list, ",")[[1]]
  if (length(otu_list) > 1) {
    for (otu in otu_list){
      freq <- otu_hits_table$Freq[gsub("\\[|\\]|\\'|\\ ", "",otu_hits_table$Var1) == otu]
      added_freq <- list_freq / length(otu_list)
      updated_freq <- freq + added_freq
      otu_hits_table$updated_freq[gsub("\\[|\\]|\\'|\\ ", "",otu_hits_table$Var1) == otu] <- updated_freq
    }
  }
}

# add Freq to updated frequencies for otus w/o multiple hits
for (otu_list in otu_hits_table$Var1[is.na(otu_hits_table$updated_freq)]) {
  list_freq <- otu_hits_table$Freq[otu_hits_table$Var1 == otu_list]
  otu_list <- gsub("\\[|\\]|\\'|\\ ", "", otu_list)
  otu_list <- strsplit(otu_list, ",")[[1]]
  if (length(otu_list) == 1) {
    otu_hits_table$updated_freq[gsub("\\[|\\]|\\'|\\ ", "",otu_hits_table$Var1) == otu_list] <- list_freq
  }
}
# make new df and format otu name
otu_hits_table <- subset(otu_hits_table, !is.na(otu_hits_table$updated_freq))
otu_hits_table$Var1 <- gsub("\\[|\\]|\\'|\\ ", "", otu_hits_table$Var1)

```

# remove Moorella thermoacetica hits from df and update otu_hits and tree
```{r}
# closest_refInfo_noMThermCopy2 <- subset(closestRefInfo_allScoreThresholdHits_singleRefOnly, grepl("copy_2", closestRefInfo_allScoreThresholdHits_singleRefOnly$closest_ref) == FALSE)
# moorella_queries <- queryDistanceInfo$query_names[grepl("copy_2", queryDistanceInfo$closest_ref) == TRUE]
# 
# ggplot(subset(closestRefInfo_allScoreThresholdHits_singleRefOnly, grepl("Archaeoglobus", closestRefInfo_allScoreThresholdHits_singleRefOnly$closest_ref) == TRUE)) +
#   geom_histogram(aes(x = distance))

tree <- phyloseq(trimmed_result_tree)
# moorella_clade <- tree_subset(tree, node = MRCA(tree, c("Moorella_thermoacetica_Y72_copy_2__MooTher5", "QUERY___S1460")), levels_back = 0)
# clade_level2 <- tree_subset(tree, node = MRCA(tree, c("Moorella_thermoacetica_Y72_copy_2__MooTher5", "QUERY___S1460")), levels_back = 2)
# 
# ggtree(clade_level2, aes(color = group)) + geom_tiplab()

# confirm that moorella_clade identifies all queries in the copy 2 clade
# now let's remove it from the tree

tree <- drop.tip(tree, moorella_clade$tip.label)

ggtree(tree, layout = "circular") +
  geom_label_repel(aes(label= ifelse(label %in% closest_ref_leaves, label, NA)), size = 1, max.overlaps = Inf, label.padding = 0.1)
```

# visualizing result tree with phyloseq and tip_glom
```{r}
# prep the tree: read in file, root, and trim to include only closest ref leaves and query leaves
# result_tree = read.tree("/Users/rebeccachristensen/Desktop/Cremer_Lab_2022/main/workflow/out/raxmlOutputFinal/RAxML_labelledTree_noBootstrap_rooted.newick")
# rooted_result_tree <-root(result_tree, c("Moorella_thermoacetica_Y72_copy_2__MooTher5"), r = TRUE)
# closest_ref_leaves <- as.vector(otu_hits_table$Var1)  # closest ref leaves from otu_hits_table (made from the ref tree)
# query_leaves <- as.vector(subset(result_tree$tip.label, grepl("QUERY", result_tree$tip.label) == TRUE))
# trimmed_result_tree <- keep.tip(result_tree, c(closest_ref_leaves, query_leaves))
# 
# # remove moorella copy 2 from the tree
# moorella_clade <- tree_subset(trimmed_result_tree, node = MRCA(trimmed_result_tree, c("Moorella_thermoacetica_Y72_copy_2__MooTher5", "QUERY___S1460")), levels_back = 0)
# trimmed_result_tree <- drop.tip(trimmed_result_tree, moorella_clade$tip.label)

# 
tree <- phyloseq(trimmed_result_tree)
# 

# subsetting by Bilophila
# tree <- tree_subset(trimmed_result_tree, "Bilophila_wadsworthia__BilWads6", levels_back = 4)
# plot_tree(subset, label.tips = "taxa_names")
# plot_tree(tip_glom(subset, h = 0.8), label.tips = "taxa_names")

# labeling reference sequences
# plot_tree(subset)

tree <- 
  ggtree(tree, layout = "circular") +
  # geom_label_repel(aes(label= ifelse(label %in% closest_ref_leaves, label, NA)), size = 1, max.overlaps = Inf, label.padding = 0.1, nudge_x = 2)
  geom_tiplab2(aes(subset=(label %in% closest_ref_leaves)), align = TRUE, size = 2)

tree %<+%
  data +
  aes(color = group) +
  scale_color_manual(values = c("red", "blue", "orange")) +
  guides(color = "none")

```


# categorize leaves as query or ref (for aesthetic mapping)
```{r}
tree <- phyloseq(trimmed_result_tree)
data <- data_frame("tip"= tree$tip.label)
data$ref_or_query <- ifelse(grepl("QUERY", data$tip), "query", "ref")

desulfovibronaceae_clade <- tree_subset(tree, node = MRCA(tree, desulfovibrionaceae_node), levels_back = 0)
firmicutes_clade <- tree_subset(tree, node = MRCA(tree, firmicutes_node), levels_back = 0)
# archaeoglobus_clade <- tree_subset(tree, node = MRCA(tree, archaeoglobus_node), levels_back = 2)

data$group <- ifelse(data$tip %in% desulfovibronaceae_clade$tip.label, "Desulfovibrionaceae", 
                     ifelse(data$tip %in% firmicutes_clade$tip.label, "Firmicutes group", "other"))
```

```{r}

pdf("phylogeny_tree_to_see_label.pdf")
ggtree(groupOTU(closest_hits_tree, groupInfo), aes(fill = group, color = group)) %<+%
  tax_df_closest_hits +
  geom_tiplab(size = 2.5, offset = 0.1) +
  geom_tippoint(aes(shape = source), size = 2) +
  xlim_tree(5)
#   geom_facet(data = data.frame(otu_hits_table), panel = "N Hits", geom = geom_bar, mapping = (aes(x = Freq)), orientation = "y", stat = "identity") +
#   geom_facet(data = subset(data.frame(sample_hits_table), !is.na(single_closest_ref)), panel = "N Samples", geom = geom_bar, mapping = (aes(x = n_samples)), orientation = "y", stat = "identity") +
# guides(fill = guide_legend(title = "family-level dsrAB taxon", nrow = 3, title.position = "top"), color = "none", shape = guide_legend(title = "sequence source", nrow = 2, title.position = "top")) +
#   theme_classic() +
#   theme(legend.position = "bottom", 
#         legend.text = element_text(size = 8),
#         legend.title = element_text(size = 9),
#         axis.text.y = element_blank(), 
#         axis.ticks.y = element_blank())
```

